cron/internal/worker/main.go

```diff
@@ -284,23 +279,31 @@ func processRequest(ctx context.Context,
 	return nil
 }
 
-func startMetricsExporter() (monitoring.Exporter, error) {
+func startMetricsExporter() error {
 	exporter, err := monitoring.GetExporter()
 	if err != nil {
-		return nil, fmt.Errorf("error during monitoring.GetExporter: %w", err)
+		return fmt.Errorf("monitoring.GetExporter: %w", err)
 	}
-	if err := exporter.StartMetricsExporter(); err != nil {
-		return nil, fmt.Errorf("error in StartMetricsExporter: %w", err)
+
+	meterProvider, err := monitoring.NewMeterProvider(exporter)
+	if err != nil {
+		return fmt.Errorf("startMetricsExporter: %w", err)
 	}
 
-	if err := view.Register(
-		&stats.CheckRuntime,
-		&stats.CheckErrorCount,
-		&stats.OutgoingHTTPRequests,
-		&githubstats.GithubTokens); err != nil {
-		return nil, fmt.Errorf("error during view.Register: %w", err)
+	// Set the global meter provider.
+	otel.SetMeterProvider(meterProvider)
+
+	// Handle shutdown properly so nothing leaks.
+	defer func() {
+		if err := meterProvider.Shutdown(context.Background()); err != nil {
+			fmt.Printf("couldn't shutdown meterProvider: %v\n", err)
+		}
+	}()
```

Comment:
wouldn't this close the meter provider as soon as the function returns? 

I think the Go idiom you're looking forward is returning a cancel/close/cleanup function, and then the caller can call/defer that when appropriate.